runtime/Cpp/cmake/ExternalAntlr4Cpp.cmake

```diff
@@ -1,223 +1,131 @@
-  antlr4cpp
-  #--Depend-on-antrl-tool-----------
-  # DEPENDS antlrtool
-  #--Core-directories-----------
-  PREFIX             ${ANTLR4CPP_EXTERNAL_ROOT}
-  #--Download step--------------
-  GIT_REPOSITORY     ${ANTLR4CPP_EXTERNAL_REPO}
-  # GIT_TAG          ${ANTLR4CPP_EXTERNAL_TAG}
-  TIMEOUT            10
-  LOG_DOWNLOAD       ON
-  #--Update step----------
-  UPDATE_COMMAND     ${GIT_EXECUTABLE} pull
-  #--Patch step----------
-  # PATCH_COMMAND sh -c "cp <SOURCE_DIR>/scripts/CMakeLists.txt <SOURCE_DIR>"
-  #--Configure step-------------
-  CONFIGURE_COMMAND  ${CMAKE_COMMAND} -DCMAKE_BUILD_TYPE=Release -DANTLR4CPP_JAR_LOCATION=${ANTLR4CPP_JAR_LOCATION} -DBUILD_SHARED_LIBS=ON -BUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR> -DCMAKE_SOURCE_DIR:PATH=<SOURCE_DIR>/runtime/Cpp <SOURCE_DIR>/runtime/Cpp
-  LOG_CONFIGURE ON
-  #--Build step-----------------
-  # BUILD_COMMAND ${CMAKE_MAKE_PROGRAM}
-  LOG_BUILD ON
-  #--Install step---------------
-  # INSTALL_COMMAND    ""
-  # INSTALL_DIR ${CMAKE_BINARY_DIR}/
-  #--Install step---------------
-  # INSTALL_COMMAND    ""
-)
-
-ExternalProject_Get_Property(antlr4cpp INSTALL_DIR)
-
-list(APPEND ANTLR4CPP_INCLUDE_DIRS ${INSTALL_DIR}/include/antlr4-runtime)
-foreach(src_path misc atn dfa tree support)
-  list(APPEND ANTLR4CPP_INCLUDE_DIRS ${INSTALL_DIR}/include/antlr4-runtime/${src_path})
-endforeach(src_path)
-
-set(ANTLR4CPP_LIBS "${INSTALL_DIR}/lib")
-
-# antlr4_shared ${INSTALL_DIR}/lib/libantlr4-runtime.so
-# antlr4_static ${INSTALL_DIR}/lib/libantlr4-runtime.a
-
-############ Generate runtime #################
-# macro to add dependencies to target
-#
-# Param 1 project name
-# Param 1 namespace (postfix for dependencies)
-# Param 2 Lexer file (full path)
-# Param 3 Parser File (full path)
-#
-# output
-#
-# antlr4cpp_src_files_{namespace} - src files for add_executable
-# antlr4cpp_include_dirs_{namespace} - include dir for generated headers
-# antlr4cpp_generation_{namespace} - for add_dependencies tracking
-
-macro(antlr4cpp_process_grammar
-    antlr4cpp_project
-    antlr4cpp_project_namespace
-    antlr4cpp_grammar_lexer
-    antlr4cpp_grammar_parser)
-
-  if(EXISTS "${ANTLR4CPP_JAR_LOCATION}")
-    message(STATUS "Found antlr tool: ${ANTLR4CPP_JAR_LOCATION}")
+if(MSVC)
+  set(ANTLR4_STATIC_LIBRARIES
+      ${ANTLR4_OUTPUT_DIR}/antlr4-runtime-static.lib)
+  set(ANTLR4_SHARED_LIBRARIES
+      ${ANTLR4_OUTPUT_DIR}/antlr4-runtime.lib)
+  set(ANTLR4_RUNTIME_LIBRARIES
+      ${ANTLR4_OUTPUT_DIR}/antlr4-runtime.dll)
+else()
+  set(ANTLR4_STATIC_LIBRARIES
+      ${ANTLR4_OUTPUT_DIR}/libantlr4-runtime.a)
+  if(MINGW)
+    set(ANTLR4_SHARED_LIBRARIES
+        ${ANTLR4_OUTPUT_DIR}/libantlr4-runtime.dll.a)
+    set(ANTLR4_RUNTIME_LIBRARIES
+        ${ANTLR4_OUTPUT_DIR}/libantlr4-runtime.dll)
+  elseif(CYGWIN)
+    set(ANTLR4_SHARED_LIBRARIES
+        ${ANTLR4_OUTPUT_DIR}/libantlr4-runtime.dll.a)
+    set(ANTLR4_RUNTIME_LIBRARIES
+        ${ANTLR4_OUTPUT_DIR}/cygantlr4-runtime-4.7.1.dll)
```

Comment:
Why's here a version number used for the binary? Should be named like the other libraries.

Reply:
CMake for Antlr4 under Cygwin generates cygantlr4-runtime-4.7.1.dll instead of cygantlr4-runtime.dll due to the declaration of SOVERSION. I just tested that building still works even if version isn't included, or IMPORTED_LOCATION is not set at all. However, it will mean the documentation in README.md has to change as I claim that ANTLR4_RUNTIME_LIBRARIES is the path to shared runtime library.

Reply:
Will you do that change yet?